tests/include/test/constant_flow.h

```diff
@@ -63,6 +76,9 @@
 #define TEST_CF_PUBLIC  __msan_unpoison
 // void __msan_unpoison(const volatile void *a, size_t size);
 
+#define TEST_CF_SAVE_SECRET(_x)    int _test_cf_is_public_ ## _x = __msan_test_shadow(&(_x), sizeof(_x)) == -1; TEST_CF_PUBLIC(&(_x), sizeof(_x));
```

Comment:
Ok, this avoids the problem I was mentioning before, but at the price of some complexity, and I'm not sure how stable that msan API is. I'm still inclined to think we should disable asm with memsan: C-level instrumentation is the wrong tool to test functions that include assembly, for that we should use binary-level instrumentation, which we have with valgrind.

Reply:
I like the idea of having Memsan be able to test as much as possible, but I agree it's a bit awkward. @tom-cosgrove-arm do you have a view on this?

Reply:
Discussed in the team sync: @gilles-peskine-arm points out that it's most important to have Memsan run against the C implementation. Since the only differences when enabling the assembly implementation are some (4) functions that become entirely assembly with no C, testing the assembly implementation with Memsan actually doesn't add a whole lot of value. So maybe reverting to the previous approach is the better option.

For valgrind, it is possibly worthwhile running `test_suite_constant_time` against the assembly implementations, but this would significantly slow down the CI just to test code that will rarely if ever get updated. So we might prefer to just rely on review. We can discuss this more in the assembly PRs.